### PR TITLE
planner: make some access paths check fit the clustered index

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/projects/45#card-42592829

Problem Summary: as the title says.

### What is changed and how it works?

What's Changed: make some access paths check fit the clustered index

### Check List 

Tests 

- Unit test

### Release note 

- none.

### CODE REVIEW

File: planner/core/logical_plans.go [hunk 1]
```diff
@@ -634,7 +634,7 @@ func (ds *DataSource) Convert2Gathers() (gathers []LogicalPlan) {
 	tg := ds.buildTableGather()
 	gathers = append(gathers, tg)
 	for _, path := range ds.possibleAccessPaths {
-		if !path.IsTablePath() {
```
Comment (winoros) [inline planner/core/logical_plans.go:637]: It was a bug?

Comment (lzmhhh123) [reply planner/core/logical_plans.go:637]: Yes, but cascades planner isn't compatible with the cluster index, so it is not an important change.

**Comment by coocood** [comment]
Can we construct some test cases that fails before the change?

**Comment by lzmhhh123** [comment]
> Can we construct some test cases that fails before the change?
Done.

**Comment by coocood** [comment]
LGTM

**Comment by coocood** [comment]
@winoros PTAL

**Pull request merged**